src/camlsnark_c/camlsnark_linker_flags_gen.ml

```diff
@@ -22,7 +22,9 @@ let () =
         ; "-lgomp"
         ; "-lssl"
         ; "-lcrypto"
-        ; "-lsodium"
+        ; "-Wl,--push-state,-Bstatic"
+        ;  "-lsodium"
+        ; "-Wl,--pop-state"
```

Comment:
We dynamically link `libsodium` on Macs in coda too, right?

Reply:
Indeed, this is the case.

Reply:
right, see `src/external/ocaml-sodium/myocamlbuild.ml` in coda.

Reply:
https://github.com/CodaProtocol/coda/blob/develop/src/external/ocaml-sodium/myocamlbuild.ml#L11